Meridian59/Data/DataController.cs

```diff
@@ -1902,6 +1913,52 @@ protected void HandleMessage(MessageMessage Message)
             ChatMessages.Add(Message.Message);           
         }
 
+        public void HandleAdminMessage(AdminMessage Message)
+        {
+            if (Message.Message.StartsWith(":< OBJECT"))
+            {
+                HandleAdminShowObjectMessage(Message);
+            }
+        }
+
+        protected void HandleAdminShowObjectMessage(AdminMessage Message)
+        {
+            //Look for object info
+            var regex = new Regex(@":< OBJECT (?<objectnumber>\d*) is CLASS (?<classname>.*)");
+            if (regex.IsMatch(Message.Message))
+            {
+
+
+            }
+            var matches = regex.Matches(Message.Message);
+            if (matches.Count < 1) //We should only get one of these at a time
+            {
+                return;
+            }
+            AdminObject obj = null;
+            foreach (Match match in matches) //but i dont know how to just do if -match will research
+            {
+                obj = new AdminObject(match.Groups["classname"].ToString(), Convert.ToInt32(match.Groups["objectnumber"].ToString()));
+            }
+
+            regex = new Regex(@": (?<property>\w*)\s*=\s(?<datatype>[\w$]*)\s(?<value>\d*)");
+            if (regex.IsMatch((Message.Message)))
+            {
+                BaseList<AdminObjectProperty> props = new BaseList<AdminObjectProperty>();
+                matches = regex.Matches(Message.Message);
+                foreach (Match match in matches)
+                {
+                    props.Add(new AdminObjectProperty(match.Groups["property"].ToString(), match.Groups["datatype"].ToString(), match.Groups["value"].ToString()));
+                }
+                if (obj != null)
+                {
+                    obj.SetProperties(props);
+                    AdminData.WatchObject(obj);
```

Comment:
I'm not exactly sure if I get this WatchObject() call and the event correctly, but I think I would prefer if:

(a) Instead your AdminData class would have a property named like "WatchedObject" of type "AdminObject".
This property would represent the currently "watched" object/the data that should be shown in something like "detailview". Your model/datacontroller code then sets this property, like "AdminData.WatchObject = obj;". Your UI code should listen for the "PropertyChanged" event of "AdminData" class.

(b) So ff not already so, your AdminData class should implement the "INotifyPropertyChanged" event (see all the other models for how to implement properties then raising PropertyChanged event).